src/conn_poll.c

```diff
@@ -230,14 +239,26 @@ int conn_poll_process(conn_registry_t *registry, pfds_record_t *pfds) {
 		if (pfd->fd == INVALID_SOCKET)
 			continue;
 
-		juice_agent_t *agent = registry->agents[i - 1];
+		bool is_tcp = i % 2 == 0;
+		int agent_index = 0;
+		if (is_tcp) {
+			agent_index = (i - 2) / 2;
+		} else {
+			agent_index = (i - 1) / 2;
+		}
+
+		juice_agent_t *agent = registry->agents[agent_index];
 		if (!agent)
 			continue;
 
 		conn_impl_t *conn_impl = agent->conn_impl;
-		if (!conn_impl || conn_impl->sock != pfd->fd || conn_impl->state != CONN_STATE_READY)
+		if (!conn_impl || conn_impl->state != CONN_STATE_READY)
 			continue;
 
+		if (conn_impl->tcp_sock != pfd->fd && conn_impl->udp_sock != pfd->fd) {
```

Comment:
```suggestion
		if (conn_impl->tcp_sock != pfd->fd || conn_impl->udp_sock != pfd->fd) {
```

Reply:
This causes the tests to fail. When poll gets fired it is either a `udp` or `tcp` socket. So one side of this branch will always fail.

Reply:
You are right, I misunderstood the logic here. But then why don't you base the `is_tcp` logic on this?